src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+		String opType = inputArr[0].toLowerCase();
+		switch (opType) {
+			case "bye":
+				return new ExitCommand(inputArr);
+			case "list":
+				return new ListCommand(inputArr);
+			case "datetime":
+				return new DateTimeCommand(inputArr);
+			case "mark":
+			case "unmark":
+				return new MarkCommand(inputArr, opType.startsWith("mark"));
+			case "delete":
+				return new DeleteCommand(inputArr);
+			case "todo":
+			case "deadline":
+			case "event":
+				return new AddCommand(inputArr);
+			case "find":
+				return new FindCommand(inputArr);
+			default:
+				return new Command(inputArr);
+		}
+	}
+
+	/**
+	 * Create task from record in saved file and add it to tasklist.
+	 * 
+	 * @param input
+	 * @param tasklist
+	 * @param ui
+	 * @param storage
+	 * @param dialogContainer
+	 */
+	public void parseAndSetData(String input, TaskList tasklist, Ui ui, Storage storage, VBox dialogContainer) {
+		String inputLower = input.toLowerCase();
+		if (inputLower.isBlank()) {
+			return;
+		}
+		String[] inputArr = inputLower.split("\\|");
+		TaskType operationType = inputLower.startsWith("t")
+				? TaskType.TODO
+				: inputLower.startsWith("d")
+						? TaskType.DEADLINE
+						: TaskType.EVENT;
+
+		Task newTask = null;
+
+		switch (operationType) {
+			case TODO:
+				newTask = new Todo(inputArr[2]);
+				break;
+
+			case DEADLINE:
+				newTask = new Deadline(inputArr[2], parseDateTimeStr(inputArr[3], ui, storage, dialogContainer));
+				break;
+
+			case EVENT:
+				newTask = new Event(inputArr[2],
+						new LocalDateTime[] { parseDateTimeStr(inputArr[3], ui, storage, dialogContainer),
+								parseDateTimeStr(inputArr[4], ui, storage, dialogContainer) });
+				break;
+
+			default:
+				break;
+		}
+		if (inputArr[1].equals("1"))
+			newTask.markAsDone();
+		tasklist.add(newTask);
+	}
+
+	/**
+	 * Convert String datetime to LocalDateTime object.
+	 * 
+	 * @param dateTimeStr
+	 * @param ui
+	 * @param storage
+	 * @param dialogContainer
+	 * @return LocalDateTime
+	 */
+	public LocalDateTime parseDateTimeStr(String dateTimeStr, Ui ui, Storage storage, VBox dialogContainer) {
+		if (dateTimeStr.isBlank()) {
+			return null;
+		}
+		String[] dateTimeArr = dateTimeStr.split(" ");
+		LocalTime time = LocalTime.MIN;
+		Integer[] dateArr;
+		String errorMessage = "Oops! Datetime format needs to be specified in proper form.";
+		if (dateTimeArr.length > 1) {
+			try {
+				int hr = Integer.parseInt(dateTimeArr[1].substring(0, 2));
+				int min = Integer.parseInt(dateTimeArr[1].substring(2, 4));
+				time = LocalTime.of(hr, min);
+			} catch (NumberFormatException ex) {
+				ui.sendResponse(dialogContainer, storage, ui.createLabel(errorMessage));
+				return null;
+			}
+		}
+
+		try {
+			if (dateTimeStr.contains("/")) { // Input Format: dd/MM/YYYY
+				dateArr = Stream.of(dateTimeArr[0].split("/"))
+						.map(Integer::valueOf)
+						.toArray(Integer[]::new);
+				return LocalDateTime.of(LocalDate.of(dateArr[2], dateArr[1], dateArr[0]), time);

```

Comment:
I like how you chain stream methods in this manner which makes it look neat and aligned